### DISCUSSION TITLE
Improving Sideband message page performance

### DISCUSSION THREAD

**Discussion by SebastianObi** [markqvist/Reticulum#124] [Ideas]
Hi Mark,

It would be good if some solution could be found to improve the performance further. I did some research and found the following. Maybe this is helpful here.

In messages.py this could possibly bring an improvement. What do you think?
I just had a quick look at it.

https://kivy.org/doc/stable/api-kivy.uix.recycleview.html

`The RecycleView provides a flexible model for viewing selected sections of large data sets. It aims to prevent the performance degradation that can occur when generating large numbers of widgets in order to display many data items.`

This sounds like a solution.

**Comment by markqvist** [comment]
Thanks for bringing it up and linking to the recycleview. It does indeed seem like the solution. I already had a look at it while preparing the 0.2.5 release, but unfortunately I didn't have time to get it in there, since it requires a bit of refactoring of the message view classes.

If anyone else is up for giving it a shot, I will happily accept pull requests :)
